torch/utils/cpp_extension.py

```diff
@@ -60,15 +78,77 @@ def build_extensions(self):
         else:
             compiler = os.environ.get('CXX', 'c++')
         check_compiler_abi_compatibility(compiler)
-        for extension in self.extensions:
-            extension.extra_compile_args = ['-std=c++11']
+
+        # Register .cu and .cuh as valid source extensions.
+        self.compiler.src_extensions += ['.cu', '.cuh']
+        # Store the compiler instance on the function (to retrieve it inside).
+        _wrap_compile.compiler_object = self.compiler
```

Comment:
This is some quite ugly use of global state. Can't we make a function that returns a closure with `self.compiler`?

Reply:
We can, I just didn't want to nest another function in here. But it's probably a nicer solution.